From 4a1e3d1e344773bad29655b5d854cc8c6c637872 Mon Sep 17 00:00:00 2001
From: Alexandru Croitor <alexandru.croitor@qt.io>
Date: Wed, 10 Sep 2025 16:08:32 +0200
Subject: [PATCH 124/553] CMake: Add non-internal option names for the SBOM
 project ops

They are not public API yet, but this would be a pre-requisite.

The intent is to use them for the Qt Creator SBOM CI run.

Pick-to: 6.9 6.8
Task-number: QTBUG-134894
Change-Id: I65b596e2e4ab994a23bdcb9bf9473a8240a5f9c7
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
(cherry picked from commit 556ea2a2e44165e4b6c0d42bf30cb39f151e8d10)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 cmake/QtPublicSbomHelpers.cmake | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/cmake/QtPublicSbomHelpers.cmake b/cmake/QtPublicSbomHelpers.cmake
index 25c6c52b1b1..848b0443c67 100644
--- a/cmake/QtPublicSbomHelpers.cmake
+++ b/cmake/QtPublicSbomHelpers.cmake
@@ -356,19 +356,20 @@ function(_qt_internal_sbom_setup_project_ops)
         list(APPEND options VERIFY_SBOM_REQUIRED)
     endif()
 
-    if(QT_INTERNAL_SBOM_VERIFY_NTIA_COMPLIANT OR QT_INTERNAL_SBOM_DEFAULT_CHECKS)
+    if(QT_SBOM_VERIFY_NTIA_COMPLIANT
+            OR QT_INTERNAL_SBOM_VERIFY_NTIA_COMPLIANT OR QT_INTERNAL_SBOM_DEFAULT_CHECKS)
         list(APPEND options VERIFY_NTIA_COMPLIANT)
     endif()
 
-    if(QT_INTERNAL_SBOM_SHOW_TABLE OR QT_INTERNAL_SBOM_DEFAULT_CHECKS)
+    if(QT_SBOM_SHOW_TABLE OR QT_INTERNAL_SBOM_SHOW_TABLE OR QT_INTERNAL_SBOM_DEFAULT_CHECKS)
         list(APPEND options SHOW_TABLE)
     endif()
 
-    if(QT_INTERNAL_SBOM_AUDIT OR QT_INTERNAL_SBOM_AUDIT_NO_ERROR)
+    if(QT_SBOM_AUDIT OR QT_INTERNAL_SBOM_AUDIT OR QT_INTERNAL_SBOM_AUDIT_NO_ERROR)
         list(APPEND options AUDIT)
     endif()
 
-    if(QT_INTERNAL_SBOM_AUDIT_NO_ERROR)
+    if(QT_SBOM_AUDIT_NO_ERROR OR QT_INTERNAL_SBOM_AUDIT_NO_ERROR)
         list(APPEND options AUDIT_NO_ERROR)
     endif()
 
@@ -380,7 +381,7 @@ function(_qt_internal_sbom_setup_project_ops)
         list(APPEND options LINT_SOURCE_SBOM)
     endif()
 
-    if(QT_INTERNAL_LINT_SOURCE_SBOM_NO_ERROR)
+    if(QT_LINT_SOURCE_SBOM_NO_ERROR OR QT_INTERNAL_LINT_SOURCE_SBOM_NO_ERROR)
         list(APPEND options LINT_SOURCE_SBOM_NO_ERROR)
     endif()
 
-- 
2.51.2

